src/utilcode/loaderheap.cpp

```diff
@@ -723,6 +729,9 @@ struct LoaderHeapFreeBlock
             }
 #endif
 
+#if LOADER_HEAP_REDZONE_VALUE != 0
```

Comment:
Instead of this:
- Add flag to UnlockedLoaderHeap that suppresses all filling of the memory by LoaderAllocator (you can steal some bits from `m_flProtect`), and set this flag for flag for `SystemDomain::GetGlobalLoaderAllocator()`.
- Change `UMEntryThunk::Terminate` to overwrite the entrypoint with jump to a method that reports the collected delegate problem and makes it easy to diagnose it.

While you are on this, you can delete `FEATURE_WINDOWSPHONE` from `UMEntryThunk::CreateUMEntryThunk/Terminate`. `FEATURE_WINDOWSPHONE` has been always defined for CoreCLR